masktoken/masktoken.go

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright 2022 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package masktoken
+
+import (
+	"fmt"
+	"regexp"
+)
+
+func MaskTokenFromString(log string, token string) string {
```

Comment:
I feel a bit uncomfortable about this low-level package not returning a proper error. I would think it is up to the caller to decide what needs to happen if an error happens during the masking, instead of writing the error into the output (which does not give you anything valuable programmatically to rely upon, especially since the string is not exported).

In addition to this, it is not highlighted in the doc block as behavior either.

Reply:
You make a good point. In the notification controller, it was a default that the new error was returned. I can see how that can be different for other packages.
I will make an update and add a proper doc block.